src/uwtools/drivers/ungrib.py

```diff
@@ -0,0 +1,175 @@
+"""
+A driver for the Ungrib model.
+"""
+
+import os
+import stat
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict
+
+from iotaa import asset, dryrun, task, tasks
+
+from uwtools.config.formats.nml import NMLConfig
+from uwtools.drivers.driver import Driver
+from uwtools.strings import STR
+from uwtools.utils.tasks import file
+
+
+class Ungrib(Driver):
+    """
+    A driver for Ungrib.
+    """
+
+    _driver_name = STR.ungrib
+
+    def __init__(
+        self, config_file: Path, cycle: datetime, dry_run: bool = False, batch: bool = False
+    ):
+        """
+        The driver.
+
+        :param config_file: Path to config file.
+        :param cycle: The forecast cycle.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config_file=config_file, dry_run=dry_run, batch=batch)
+        self._config.dereference(context={"cycle": cycle})
+        if self._dry_run:
+            dryrun()
+        self._cycle = cycle
+
+    # Workflow tasks
+
+    @task
+    def gribfile_aaa(self):
+        """
+        The gribfile.
+        """
+        path = self._rundir / "GRIBFILE.AAA"
+        yield self._taskname(str(path))
+        yield asset(path, path.is_symlink)
+        infile = Path(self._driver_config["gfs_file"])
+        yield file(path=infile)
+        path.parent.mkdir(parents=True, exist_ok=True)
+        path.symlink_to(infile)
+
+    @task
+    def namelist_wps(self):
+        """
+        The namelist file.
+        """
+        d = {
+            "update_values": {
+                "share": {
+                    "end_date": self._cycle.strftime("%Y-%m-%d_%H:00:00"),
+                    "interval_seconds": 1,
+                    "max_dom": 1,
+                    "start_date": self._cycle.strftime("%Y-%m-%d_%H:00:00"),
+                    "wrf_core": "ARW",
+                },
+                "ungrib": {
+                    "out_format": "WPS",
+                    "prefix": "FILE",
+                },
+            }
+        }
```

Comment:
I suggested to @elcarpenterNOAA during pairing that the namelist file for `ungrib` is simple enough that we could consider just creating it from scratch. The cycle-related values here we probably always want to supply. If there's a reason why, at least in the short term where we'll be using this strictly for MPAS, any of the other values shouldn't be hardcoded and have to be exposed for user configuration in UW YAML, that can be done. Can we YAGNI this for now?

Reply:
So we won't be allowing users to update any of these settings like we do with the other namelists? I'm not sure how I feel about that as the general pattern, but it should be fine here.

This end date will likely need to change along with changes to support LBCs.

Reply:
This is the only component namelist I've seen so far that seemed like it might not need user interaction. It'll be easy enough to add if/when we need it.

It'll be interesting to see how we decide to handle LBCs in the UW YAML: It would be _nice_ to just take a start time and some kind of time delta and calculate the rest, but there's no telling what the GFS GRIB files will be called. I don't think we can safely assume the NCEP filenames, so we'll probably have to accept either an explicit list of filenames, or some kind of template from which the filenames can be constructed.